### PR TITLE
Feature/user cru

### PR DESCRIPTION
**Pull request by Nicolas-Rodriguez-Ch**
### Summary

- refactoring functionality for the user sign-up, consistent now with the back end
- creating functionality for the login component, stores the user in a redux store
- reactoring the user page and its route, it is currently static and gets the user from the token instead of the id.
- adding toast to all the modified views and components to show success or failure messages.

### To-Do

- conditional rendering failure or missing token or token expired in the user page.
- the ability to delete user on the user page.
- conditionally show different buttons on the login modal if the user is logged in.
- privatization of admin routes and add dynamism according to the back end.

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -2,14 +2,21 @@ import "./App.css";
...
 import RestaurantAdminView from "./pages/RestaurantAdminView/RestaurantAdminView";
 import GeneralAdminView from "./pages/GeneralAdminView/GeneralAdminView";
 import { CheckoutPageComponent } from "./pages/CheckoutPageComponent/CheckoutPageComponent";
 import UserPage from "./pages/UserPage/UserPage";
+import Cookies from "universal-cookie";
+
+const Private = ({ children }) => {
+  const cookies = new Cookies();
+  const ticket = cookies.get("token");
+  return ticket ? children : <Navigate to={"/"}/>
```
Comment (ht1204) [inline src/App.js]: `return ticket ? children : <Navigate to={"/"}/>;`

File: src/components/LogInComponent/LoginFormComponent.jsx [hunk 2]
```diff
@@ -1,48 +1,99 @@
...
+
+  const handleSubmit = async (e) => {
+    e.preventDefault();
+    try {
+      const { data } = await axios.post(`${authUrl}/local/login`, dataUser);
+      cookies.set("token", data.token);
+      cookies.set("name", data.data.name);
+      cookies.set("last_name", data.data.last_name);
+      cookies.set("email", data.data.email);
+      dispatch(setUser(data.data));
```
Comment (ht1204) [inline src/components/LogInComponent/LoginFormComponent.jsx:39]:
```js
const { 
 data: {
   token,
   data: { name, last_name, email }
 }
} = await axios.post(`${authUrl}/local/login`, dataUser);
```

File: src/components/LogInComponent/LoginFormComponent.jsx [hunk 3]
```diff
@@ -1,48 +1,99 @@
...
+      toast.success(languageSelector(language, "logInSuccessfull"));
+
+      setTimeout(() => {
+        closeModal();
+      }, 6000);
```
Comment (ht1204) [inline src/components/LogInComponent/LoginFormComponent.jsx:44]: I consider avoiding any lack of memory or non desirable behavior using this kind of implementation with setTimeout:
```js
    const timerModal = setTimeout(() => {
        closeModal();
      }, 6_000);
    clearTimeout(timerModal);
```

File: src/pages/UserPage/UserPage.jsx [hunk 4]
```diff
@@ -43,16 +52,16 @@ const UserPage = () => {
...
-          email, 
-          password, 
-          phone_numbers, 
-          addresses, 
-          city 
+          last_name,
+          email,
+          password,
+          phone_numbers,
+          addresses, //try later addresses: address ,
```
Comment (ht1204) [inline src/pages/UserPage/UserPage.jsx]: is this comment necessary?

**Review by ht1204 — COMMENTED** [review]
Consider bit enhancements to be done, please address the comments.

File: src/App.js (hunk 1 above)
Comment (Nicolas-Rodriguez-Ch) [reply src/App.js]: Resolved

File: src/components/LogInComponent/LoginFormComponent.jsx (hunk 2 above)
Comment (Nicolas-Rodriguez-Ch) [reply src/components/LogInComponent/LoginFormComponent.jsx:39]: Resolved in a different way.

File: src/components/LogInComponent/LoginFormComponent.jsx (hunk 3 above)
Comment (Nicolas-Rodriguez-Ch) [reply src/components/LogInComponent/LoginFormComponent.jsx:44]: Resolved, but I used a state a useeffect to get the expected behavior

File: src/pages/UserPage/UserPage.jsx (hunk 4 above)
Comment (Nicolas-Rodriguez-Ch) [reply src/pages/UserPage/UserPage.jsx]: No, it was a comment for myself that i forgot to try it and erase it. 

Resolved

**Comment by Nicolas-Rodriguez-Ch** [comment]
Made the corresponding changes

File: src/components/LogInComponent/LoginFormComponent.jsx (hunk 2 above)
Comment (ht1204) [reply src/components/LogInComponent/LoginFormComponent.jsx:39]: However you did it well and onelinear, great, the code aesthetic is a well practice that must be part of us.

File: src/components/LogInComponent/LoginFormComponent.jsx (hunk 3 above)
Comment (ht1204) [reply src/components/LogInComponent/LoginFormComponent.jsx:44]: The refactor was better yet, using hooks was a relevant change, good one!

File: src/components/LogInComponent/LoginFormComponent.jsx [hunk 5]
```diff
@@ -13,33 +13,44 @@ const LoginFormComponent = ({ setWhichForm, closeModal }) => {
...
+      setTimerModal(timer);
     } catch (error) {
       if (error.response && error.response.data && error.response.data.message) {
         toast.error(error.response.data.message);
       } else {
         toast.error(languageSelector(language, "generalError"));
       }
     }
-  }
+  };
```
Comment (ht1204) [inline src/components/LogInComponent/LoginFormComponent.jsx:53]: semicolon not necessary

**Review by ht1204 — APPROVED** [review]

**Pull request merged**